test/core/channel/channel_stack_builder_test.cc

```diff
@@ -57,70 +53,57 @@ void CallDestroyFunc(grpc_call_element* /*elem*/,
-  g_replacement_fn_called = true;
-}
-
-void SetOriginalFnCalled(grpc_channel_stack*, grpc_channel_element*) {
-  g_original_fn_called = true;
+const grpc_channel_filter* FilterNamed(const char* name) {
+  static auto* filters =
+      new std::map<absl::string_view, const grpc_channel_filter*>;
+  auto it = filters->find(name);
+  if (it != filters->end()) return it->second;
+  return filters
+      ->emplace(
+          name,
+          new grpc_channel_filter{
+              grpc_call_next_op, nullptr, grpc_channel_next_op, 0, CallInitFunc,
+              grpc_call_stack_ignore_set_pollset_or_pollset_set,
+              CallDestroyFunc, 0, ChannelInitFunc,
+              [](grpc_channel_stack*, grpc_channel_element*) {},
+              ChannelDestroyFunc, grpc_channel_next_get_info, name})
+      .first->second;
 }
 
-TEST(ChannelStackBuilderTest, ReplaceFilter) {
-  grpc_channel_credentials* creds = grpc_insecure_credentials_create();
-  grpc_channel* channel =
-      grpc_channel_create("target name isn't used", creds, nullptr);
-  grpc_channel_credentials_release(creds);
-  GPR_ASSERT(channel != nullptr);
-  // Make sure the high priority filter has been created.
-  GPR_ASSERT(g_replacement_fn_called);
-  // ... and that the low priority one hasn't.
-  GPR_ASSERT(!g_original_fn_called);
-  grpc_channel_destroy(channel);
-}
-
-const grpc_channel_filter replacement_filter = {
-    grpc_call_next_op,    nullptr,
-    grpc_channel_next_op, 0,
-    CallInitFunc,         grpc_call_stack_ignore_set_pollset_or_pollset_set,
-    CallDestroyFunc,      0,
-    ChannelInitFunc,      SetReplacementFnCalled,
-    ChannelDestroyFunc,   grpc_channel_next_get_info,
-    "filter_name"};
-
-const grpc_channel_filter original_filter = {
-    grpc_call_next_op,    nullptr,
-    grpc_channel_next_op, 0,
-    CallInitFunc,         grpc_call_stack_ignore_set_pollset_or_pollset_set,
-    CallDestroyFunc,      0,
-    ChannelInitFunc,      SetOriginalFnCalled,
-    ChannelDestroyFunc,   grpc_channel_next_get_info,
-    "filter_name"};
-
-bool AddReplacementFilter(ChannelStackBuilder* builder) {
-  // Get rid of any other version of the filter, as determined by having the
-  // same name.
-  auto* stk = builder->mutable_stack();
-  stk->erase(std::remove_if(stk->begin(), stk->end(),
-                            [](const grpc_channel_filter* entry) {
-                              return strcmp(entry->name, "filter_name") == 0;
-                            }),
-             stk->end());
-  builder->PrependFilter(&replacement_filter);
-  return true;
+TEST(ChannelStackBuilder, UnknownTarget) {
+  ChannelStackBuilderImpl builder("alpha-beta-gamma", GRPC_CLIENT_CHANNEL,
+                                  ChannelArgs());
+  EXPECT_EQ(builder.target(), "unknown");
 }
 
-bool AddOriginalFilter(ChannelStackBuilder* builder) {
-  builder->PrependFilter(&original_filter);
-  return true;
+TEST(ChannelStackBuilder, CanPrepend) {
```

Comment:
Do we still need the `PrependFilter()` method at all anymore?  It looks like `ChannelInit::CreateStack()` is only calling `AppendFilter()`, so probably `PrependFilter()` is now unused.

Reply:
Yeah... I'd like to have it hang out for some until we've finished though: maybe we find a post-processor that really wants it. I'll add a TODO to the builder interface to re-evaluate later.